engines/chamber/kult.cpp

```diff
@@ -354,6 +353,8 @@ Common::Error ChamberEngine::run() {
 
 	/* Main game loop */
 	gameLoop(frontbuffer);
+	if (g_vm->_shouldRestart)
+		run();
```

Comment:
This is very, very bad design. You are making a recursive call to this very method. Please, avoid it.

Instead of writing like this, you would better refactor it into something like

```C++
Common::Error ChamberEngine::run() {
   init();

   do {
        g_vm->_shouldRestart = false;
        execute();
    } while (g_vm->_shouldRestart);
```

Then, put the existing contents of the ::run() method into ::execute(), splitting it into two parts: init() and execute(). The `init()` one is the code that you are currently wrapping into `if (g_vm->_shouldRestart())`, and the rest goes into `execute()`. And NO recursion!